remote-workstations/new-agent-vm/Install-PCoIPAgent.sh

```diff
@@ -243,131 +238,326 @@ active_d = ldap_lib(args.address, args.user_name, args.password, args.domain)
 active_d.add_computer_to_group(args.computer, args.group)
 
 EOF
-sudo python $file_path -d "$DOMAIN_NAME" -a "$DC_ADDRESS" -u "$USERNAME" -p "$PASSWORD" -c "$VM_NAME" -g "$GROUP"
 
+    sudo python $file_path -d "$DOMAIN_NAME" -a "$DC_ADDRESS" -u "$USERNAME" -p "$PASSWORD" -c "$VM_NAME" -g "$GROUP"
+}
+
+install_gui() 
+{
+    # Make sure Linux OS is up to date
+    echo "--> Updating Linux OS to latest"
+
+    sudo yum -y update  # --exclude=WALinuxAgent
+
+    # Install Desktop
+    echo "-->Install desktop"	
+    #sudo yum -y groupinstall 'X Window System' 'GNOME'
+    sudo yum -y groupinstall "Server with GUI"
+    
+    # sudo yum groups mark convert
+
+    # install firefox
+    echo "-->Install firefox"
+    sudo yum -y install firefox
+    
+    #echo "-->set default graphical target"
+    # The below command will change runlevel from runlevel 3 to runelevel 5 
+    sudo systemctl set-default graphical.target
+    
+    #echo "-->start graphical target"
+    sudo systemctl start graphical.target	
+}	
+
+install_pcoip_agent()
+{
+    # Install the Teradici package key
+    echo "-->Install the Teradici package key"
+    sudo rpm --import https://downloads.teradici.com/rhel/teradici.pub.gpg
+    
+    # Add the Teradici repository
+    echo "-->Add the Teradici repository"
+    
+    agent_repo_url="https://downloads.teradici.com/rhel/pcoip.repo"
+    case "$AGENT_CHANNEL" in 
+        "beta")
+            agent_repo_url="https://downloads.teradici.com/rhel/pcoip-beta.repo"
+            ;;
+        "dev")
+            agent_repo_url="https://downloads.teradici.com/rhel/pcoip-dev.repo"
+            ;;   
+        *)
+            agent_repo_url="https://downloads.teradici.com/rhel/pcoip.repo"
+            ;;       
+    esac
+    
+    sudo wget --retry-connrefused --tries=3 --waitretry=5 -O /etc/yum.repos.d/pcoip.repo $agent_repo_url
```

Comment:
The `$agent_repo_url` variable should be in double quotes.

Reply:
using double quotes